pkgs/build-support/testers/default.nix

```
@@ -9,10 +9,18 @@
       version ? package.version,
     }: runCommand "${package.name}-test-version" { nativeBuildInputs = [ package ]; meta.timeout = 60; } ''
       if output=$(${command} 2>&1); then
-        grep -Fw "${version}" - <<< "$output"
-        touch $out
+        if grep -Fw "${version}" - <<< "$output"; then
+          touch $out
+        else
+          echo "Version string '${version}' not found!" >&2
+          echo "The output was:" >&2
+          echo "$output" >&2
+          exit 1
+        fi
       else
-        echo "$output" >&2 && exit 1
+        echo "'${command}' returned a non-zero exit code." >&2
```

Probably need something like:

```suggestion
        echo "'"${lib.escapeShellArg command}"' returned a non-zero exit code." >&2
```

---

I think you really need to move it out of the bash string, the function only escapes single quotes – and even if it escaped double quotes, you still would be relying on implementation details, which is no good.